### PR TITLE
PMM-6279 Show query metadata in the details.

### PR DESCRIPTION
**Pull request by IuriiMr**
PMM-6279

Feature Build:
 - https://github.com/Percona-Lab/pmm-submodules/pull/3137

Related PRs: 
- https://github.com/percona/pmm/pull/1865

### CODE REVIEW

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.styles.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export const getStyles = (theme: GrafanaTheme2) => {
+  const parameters = getPmmTheme(theme.v1);
+
+  return {
+    metadataColumn: css`
+      display: flex;
+      justify-content: space-between;
+      align-items: center;
+      word-break: keep-all;
+      color: ${parameters.mainTextColor} !important;
```
Comment (ritbl) [inline pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.styles.ts:14]: can we avoid `!important`?

Comment (ritbl) [reply pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.styles.ts:14]: these will be inline styles, I don't really understand why it is needed.

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import { Overlay } from 'shared/components/Elements/Overlay/Overlay';
+import { Collapse, useStyles2 } from '@grafana/ui';
+import { Table } from '../../../../../shared/components/Elements/Table';
+import { LinkTooltip } from '../../../../../shared/components/Elements/LinkTooltip/LinkTooltip';
+import { getStyles } from './Metadata.styles';
+import { metadataTableHead } from './Metadata.constants';
+
+const Metadata = ({ metadata, loading }) => {
+  const styles = useStyles2(getStyles);
+  const [isMetadataOpen, setMetadataVisibility] = useState(true);
```
Comment (ritbl) [inline pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx:11]: nit: usually it is called `[<name>, set<name>]`, for example [metadataOpen, setMetadataOpen]

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      accessor: nameColumn,
+    },
+    {
+      Header: metadataTableHead.second,
+      accessor: valueColumn,
+    },
+  ];
+
+  return (
+    <Overlay size={35}>
```
Comment (ritbl) [inline pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx:38]: why 35?

**Review by ritbl — APPROVED** [review]
Looks good, left few nits here and there

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.styles.ts (hunk 1 above)
Comment (IuriiMr) [reply pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.styles.ts:14]: Yes, we can avoid this, but in this case I've kept the style of the table 'Metrics' above.

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import { Overlay } from 'shared/components/Elements/Overlay/Overlay';
+import { Collapse, useStyles2 } from '@grafana/ui';
+import { Table } from '../../../../../shared/components/Elements/Table';
```
[the remark this answers is not in the document]
Comment (IuriiMr) [reply pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx:4]: The paths were changed to common in the project.

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx (hunk 3 above)
Comment (IuriiMr) [reply pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx:38]: I've kept the style of the table 'Metrics' above to have the same size.

**Review by ritbl — APPROVED** [review]

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.constants.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
+// Part of tooltips from https://github.com/percona/pmm/blob/main/docs/managed/data-model.md
+export const metadataNamesTooltips: {[key: string]: {name: string, tooltip: string}} = {
+  service_name: {
+    name: 'Service name',
+    tooltip: 'Unique across all Services user-defined name, can be changed.',
```
Comment (fabio-silva) [inline pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.constants.ts:5]:
```suggestion
    tooltip: 'Unique across all Services. User-defined name, can be changed.',
```

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.constants.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    name: 'Database',
+    tooltip: 'Database.',
+  },
+  schema: {
+    name: 'Schema',
+    tooltip: 'Schema.',
+  },
+  username: {
+    name: 'Username',
+    tooltip: 'Username for scraping metrics / Authentication user.',
```
Comment (fabio-silva) [inline pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.constants.ts:17]: Why is "Authentication" camel cased?

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    },
+  ];
+
+  return (
+    <Overlay size={OVERLAY_LOADER_SIZE}>
+      <Collapse
+        collapsible
+        label={metadataTableHead.main}
+        isOpen={metadataOpen}
+        onToggle={() => setMetadataOpen(!metadataOpen)}
```
Comment (fabio-silva) [inline pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx:44]:
```suggestion
        onToggle={() => setMetadataOpen((open) => !open)}
```

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+      >
+        <Table
+          columns={columns}
+          data={metadata}
+          loading={loading}
+          noData={null}
```
Comment (fabio-silva) [inline pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx:50]: I believe this is an optional prop, so there's no need to send null

**Review by fabio-silva — CHANGES_REQUESTED** [review]

**Review by solovevayaroslavna — COMMENTED** [review]
it looks good to me, except for comments left by colleagues

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.constants.ts (hunk 5 above)
Comment (IuriiMr) [reply pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.constants.ts:5]: Fixed.

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.constants.ts (hunk 6 above)
Comment (IuriiMr) [reply pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.constants.ts:17]: Fixed.

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx (hunk 7 above)
Comment (IuriiMr) [reply pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx:44]: Changed.

File: pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx (hunk 8 above)
Comment (IuriiMr) [reply pmm-app/src/pmm-qan/panel/components/Details/Metadata/Metadata.tsx:50]: Removed an optional.

**Comment by IuriiMr** [comment]
All tooltips have been removed based on design review by Pedro Fernandes from 03/27/2023.

**Review by fabio-silva — APPROVED** [review]

**Review by ritbl — APPROVED** [review]
LGTM

**Pull request merged**